### PR TITLE
Devops-3259 

### PR DESCRIPTION
**Pull request by HarshitaInfoblox**
As per comment in PR #12 . Created this against helm3 branch

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+pipeline {
+  agent {
+    label 'ubuntu_docker_label'
+  }
+  options {
+    checkoutToSubdirectory('src/github.com/infobloxopen/helm')
+  }
+  environment {
+    DIRECTORY = "src/github.com/infobloxopen/helm"
+  }
```
Comment (darrowmire) [inline Jenkinsfile]: You can remove this, it's needed only for Golang based repos

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  stages {
+    stage("Setup") {
+      steps {
+        // prepareBuild is one of the Secure CICD helper methods
+        prepareBuild()
+      }
+    }
+    stage("Build Image") {
+      steps {
+        dir("$DIRECTORY") {
```
Comment (darrowmire) [inline Jenkinsfile]: you can remove dir block too

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+      }
+    }
+    stage("Push Image") {
+      //when { expression { ! isPrBuild() }  }
+      when { anyOf { branch "master", branch "helm3" } }
+      steps {
+        // reference Jenkins credential ids via an environment variable
+        withDockerRegistry([credentialsId: "${env.JENKINS_DOCKER_CRED_ID}", url: ""]) {
+          dir("$DIRECTORY") {
```
Comment (darrowmire) [inline Jenkinsfile]: comment above

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+             sh "make docker-push"
+          }
+        }
+      }
+    }
+  post {
+    success {
+      // finalizeBuild is one of the Secure CICD helper methods
+      dir("$DIRECTORY") {
```
Comment (darrowmire) [inline Jenkinsfile]: comment above

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          finalizeBuild()
+      }
+    }
+    cleanup {
+      dir("$DIRECTORY") {
```
Comment (darrowmire) [inline Jenkinsfile]: comment above

**Review by darrowmire — APPROVED** [review]

File: Makefile [hunk 6]
```diff
@@ -5,9 +5,17 @@ default: Dockerfile
 	docker build -t infoblox/helm:$(IMAGE_VERSION) .
 	docker push infoblox/helm:$(IMAGE_VERSION)
 
+build: Dockerfile
```
Comment (tgjonestx) [inline Makefile:6]: [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) concern - please remove recipe for the `default` target and change to `default: build docker-push`

File: Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        prepareBuild()
+      }
+    }
+    stage("Build Image") {
+      steps {
+           sh "make build"
+      }
+    }
+    stage("Push Image") {
+      //when { expression { ! isPrBuild() }  }
```
Comment (tgjonestx) [inline Jenkinsfile]: did this check for `!isPrBuild()` not work?

File: Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  }
+  post {
+    success {
+      // finalizeBuild is one of the Secure CICD helper methods
+          finalizeBuild()
+    }
+    cleanup {
+        sh "make clean || true"
+      cleanWs()
```
Comment (tgjonestx) [inline Jenkinsfile:37]: Please fixup indent/spacing issues for both `success` and `cleanup`

**Review by tgjonestx — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 7 above)
Comment (HarshitaInfoblox) [reply Jenkinsfile]: Thought that we'd want to push only from helm3 & master branch. But `!isPrBuild` also makes sense. Should I change to `!isPrBuild`?

Comment (tgjonestx) [reply Jenkinsfile]: Please try !isPrBuild

**Review by tgjonestx — APPROVED** [review]

**Pull request merged**